### DISCUSSION TITLE
Why require "using" to be at the top?

### DISCUSSION THREAD

**Discussion by God-damnit-all** [PowerShell/PowerShell#20047] [Q&A]
Some time ago, I discovered that it's possible to add the use of a namespace anywhere through a command like
`. ([ScriptBlock]::Create('using namespace System.Collections'))` and I'm happier for it.

But, that makes me wonder, why is it that PowerShell's interpreter throws a fit whenever it's not at the top in the first place if this workaround appears to work just fine?

**Comment by IISResetMe** [comment]
It's a design choice, likely to communicate that "this is not a regular statement".

`using` statements, along with type definitions (`enum ...` and `class ...`) are processed and compiled during _parsing_, so having a `using ...` statement in the middle of a file would be confusing.
